bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,313 @@
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link QbusBridgeHandler} is the handler for a Qbus controller
+ *
+ * @author Koen Schockaert - Initial Contribution
+ */
+
+@NonNullByDefault
+public class QbusBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(QbusBridgeHandler.class);
+
+    private @Nullable QbusCommunication qbusComm;
+
+    protected @Nullable QbusConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshTimer;
+
+    public QbusBridgeHandler(Bridge Bridge) {
+        super(Bridge);
+    }
+
+    /**
+     * Initialize the bridge
+     */
+    @Override
+    public void initialize() {
+        readConfig();
+
+        InetAddress addr;
+        Integer port = getPort();
+        Integer refresh = getRefresh();
+
+        if (port == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "No port defined for Qbus Server");
+            return;
+        }
+
+        try {
+            addr = InetAddress.getByName(getAddress());
+            if (addr == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                        "No ip address defined for Qbus Server");
+                return;
+            } else {
+                createCommunicationObject(addr, port);
+            }
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Incorrect ip address set for Qbus Server");
+        }
+
+        if (refresh != null) {
+            this.setupRefreshTimer(refresh);
+        }
+    }
+
+    /**
+     * Sets the Bridge call back
+     */
+    private void setBridgeCallBack() {
+        QbusCommunication qbusCommunication = getQbusCommunication();
+        if (qbusCommunication != null) {
+            qbusCommunication.setBridgeCallBack(this);
+        }
+    }
+
+    /**
+     * Create communication object to Qbus server and start communication.
+     *
+     * @param addr : IP address of Qbus server
+     * @param port : Communication port of QbusServer
+     */
+    private void createCommunicationObject(InetAddress addr, int port) {
+        scheduler.submit(() -> {
+
+            setQbusCommunication(new QbusCommunication());
+
+            QbusCommunication qbusCommunication = getQbusCommunication();
+
+            setBridgeCallBack();
+            if (qbusCommunication != null) {
+                try {
+                    qbusCommunication.startCommunication();
+                } catch (InterruptedException | IOException e) {
+                    logger.warn("Error on restaring communication: {}", e.getMessage());
+                    bridgeOffline("Communication could not be established");
```

Comment:
The log message could be removed, as the status update is already logged by the framework.

The state change originated by updateStatus() is logged to events.log. Including the status detail message.

Also, it's good practice to include the exception's message in the status detail message to display it in the UI.